### PR TITLE
Merge ocp47-2021-2 branch and add scripts for secrets and certs

### PR DESCRIPTION
**Pull request by hollisc**
This PR contains the following updates: 

- Merge changes from `ocp47-2021-2` branch into `master`
- Add scripts to generate `certificate` resources with cluster domain set
- Add scripts to generate jks keystore password required by the client certificate

### CODE REVIEW

**Comment by osowski** [comment]
Overall this looks like a great start. A few comments:
1. As discussed on Slack, let's mirror Callum's pattern [here](https://github.ibm.com/CALLUMJ/MQonCP4I/blob/master/resources/cp4i2020.2/externalqmgr.921/deploy/install.sh) for templating and dynamic injection into template YAMLs.
2. The above fix should handle this, but let's make sure we parameterize `$CLUSTER_DOMAIN` everywhere instead of `<cluster-domain>`.
3. I may have missed it in the changes, but I didn't see anywhere that was invoking the new shell scripts to generate Certificates/SealedSecrets as needed. I would expect those edits to show up in `multi-tenancy-gitops-apps/blob/ocp47-2021-2/scripts/mq-kubeseal.sh` if I am understanding the layout correctly.

**Comment by hollisc** [comment]
Thanks @osowski for the feedback.  Updates on some items listed below.

- Item 2 - Addressed in this commit (https://github.com/cloud-native-toolkit-demos/multi-tenancy-gitops-apps/pull/85/commits/569e6db762ddcb0681b9478fcfc03d7d3e0b0a44)
- Item 4 - Addressed in this commit (https://github.com/cloud-native-toolkit-demos/multi-tenancy-gitops-apps/pull/85/commits/569e6db762ddcb0681b9478fcfc03d7d3e0b0a44)
- Item 5 - Addressed in this commit (https://github.com/cloud-native-toolkit-demos/multi-tenancy-gitops-apps/pull/85/commits/ebf470f740b78db8408118612e40d937334ba99e)
- Item 6 -  Addressed in this commit (https://github.com/hollisc/multi-tenancy-gitops-apps/commit/7a022ef72b2676182f3f3b5cc6aaa41c1829fc8e).

**Comment by hollisc** [comment]
@osowski , item 1 and 3 have been addressed in the latest set of commits.  As discussed, for the scripts generating sealedsecret yamls, I have left it with `oc create secrets` as it is run with `--dry-run=client` to generate a yaml which gets sealed and than we delete the k8s secret.

**Comment by hollisc** [comment]
Last commit of this PR has the following changes which is needed to run promotional pipelines: 

- Add clusterrole and rolebindings for pipeline sa in `ci`
- Add argocd application to add scc to pipeline sa 
- Update argocd apps for `mq-infra` so it is part of the `applications` ArgoCD AppProject instead of `services`

File: scripts/set-git-source.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ROOTDIR="${SCRIPTDIR}/.."
+[[ -n "${DEBUG:-}" ]] && set -x
+
+
+if [ -z ${GIT_ORG} ]; then echo "Please set GIT_ORG when running script, optional GIT_BASEURL and GIT_REPO to formed the git url GIT_BASEURL/GIT_ORG/*"; exit 1; fi
+
+
+GIT_HOST=${GIT_HOST:-github.com}
+GIT_BASEURL="https://${GIT_HOST}"
+GITOPS_REPO=${GITOPS_REPO:-multi-tenancy-gitops-apps}
```
Comment (csantanapr) [inline scripts/set-git-source.sh:15]: I updated the variables name at upstream bootstrap.sh to specify apps git repo `GIT_GITOPS_APPLICATIONS`  and branch `GIT_GITOPS_APPLICATIONS_BRANCH`
```
GIT_GITOPS_APPLICATIONS=${GIT_GITOPS_APPLICATIONS:-multi-tenancy-gitops-apps}
GIT_GITOPS_APPLICATIONS_BRANCH=${GIT_GITOPS_APPLICATIONS_BRANCH:-master}
```

Comment (csantanapr) [reply scripts/set-git-source.sh:15]: Is here https://github.com/cloud-native-toolkit-demos/multi-tenancy-gitops-apps/blob/master/scripts/ace-bootstrap.sh#L28
and here 
https://github.com/cloud-native-toolkit/multi-tenancy-gitops/blob/master/scripts/bootstrap.sh#L54-L61

Comment (hollisc) [reply scripts/set-git-source.sh]: Thanks, updated the set-git-source.sh to use the new env var (`GIT_GITOPS_APPLICATIONS`and `GIT_GITOPS_APPLICATIONS_BRANCH`).

**Review by osowski — APPROVED** [review]
lgtm

**Pull request merged**